Add test suite for bottom navigation

amitojsingh opened the pull request:
I have added a test suite for bottom navigation.

amitojsingh committed:
Update ESLint configuration, remove jest.confg.js, and enhance BottomNavigation accessibility

- Added ESLint override rules to disable the jsx-filename-extension rule for test files.
- Removed the obsolete jest.confg.js file to streamline project configuration.
- Enhanced BottomNavigation component by adding accessibility roles and labels for better screen reader support.
- Updated package.json to include @testing-library/react-native for improved testing capabilities.

amitojsingh committed:
added Github action for PR

amitojsingh committed:
update branch

amitojsingh committed:
fix eslint bugs

amitojsingh committed:
Merge branch 'dev' into add-test-suite

Gauravjeetsingh commented on jest.config.js:
```diff
@@ -0,0 +1,18 @@
+    "^@react-native/js-polyfills/error-guard$":
+      "<rootDir>/__mocks__/@react-native/js-polyfills/error-guard.js",
+    "^@common/(.*)$": "<rootDir>/src/common/$1",
+    "^@database/(.*)$": "<rootDir>/src/database/db/$1",
```
Is this correct?

Based on the babel.config.js, @database maps directly to ./src/database/db (which resolves to db.js), not a subdirectory.

amitojsingh replied:
fixed

Gauravjeetsingh reviewed (changes requested):
Looks good, only minor changes for this one.

- Can we setup a jest.setup.js file and include all the operations to run before tests in that.
- Can we also add the coverage configuration. This can be done in a separate PR as well.
https://jestjs.io/docs/configuration#collectcoveragefrom-array

amitojsingh committed:
Merge branch 'dev' into add-test-suite

amitojsingh commented:
I have added jest setup file in this PR. 
https://github.com/KhalisFoundation/sundar-gutka-react/pull/324/files#diff-f1995b7576ce098f06844c16df12561f4694c56c84869ec5180933e652f56862

amitojsingh committed:
Update Jest configuration to simplify database path mapping

- Changed the Jest module path for @database from a specific file to the directory, allowing for more flexible imports in tests.

amitojsingh commented:
Sure I’ll create Seperate PR for coverage configuration.

amitojsingh committed:
Merge branch 'dev' into add-test-suite

amitojsingh committed:
Enhance BottomNavigation functionality and Jest configuration (#324)

- Added setupFilesAfterEnv to jest.config.js for improved testing setup.
- Removed BottomNavigation from Bookmarks component to streamline navigation.
- Enhanced BottomNavigation to conditionally hide navigation items based on the current route, specifically when on the Settings page.
- Updated BottomNavigation tests to verify the new behavior and ensure audio toggling functionality works as expected.
- Adjusted styles in BottomNavigation for better layout consistency.
